tron/serialize/runstate/dynamodb_state_store.py

```diff
@@ -87,14 +87,18 @@ def _get_items(self, table_keys: list) -> object:
             for resp in concurrent.futures.as_completed(responses):
                 items.extend(resp.result()["Responses"][self.name])
                 # add any potential unprocessed keys to the thread pool
-                if resp.result()["UnprocessedKeys"].get(self.name) and attempts_to_retrieve_keys < MAX_ATTEMPTS:
-                    cand_keys_list.append(resp.result()["UnprocessedKeys"][self.name]["Keys"])
-                elif attempts_to_retrieve_keys >= MAX_ATTEMPTS:
-                    failed_keys = resp.result()["UnprocessedKeys"][self.name]["Keys"]
-                    error = Exception(
-                        f"tron_dynamodb_restore_failure: failed to retrieve items with keys \n{failed_keys}\n from dynamodb\n{resp.result()}"
-                    )
-                    raise error
+                try:
+                    if resp.result()["UnprocessedKeys"].get(self.name) and attempts_to_retrieve_keys < MAX_ATTEMPTS:
+                        cand_keys_list.append(resp.result()["UnprocessedKeys"][self.name]["Keys"])
+                    elif attempts_to_retrieve_keys >= MAX_ATTEMPTS:
+                        failed_keys = resp.result()["UnprocessedKeys"][self.name]["Keys"]
+                        error = Exception(
+                            f"tron_dynamodb_restore_failure: failed to retrieve items with keys \n{failed_keys}\n from dynamodb\n{resp.result()}"
+                        )
+                        raise error
+                except Exception as e:
+                    log.error(repr(e))
```

Comment:
log.exception() will let us log a human-readable message and automatically include the trace back after said message

Reply:
it also seems like we don't need a try-except here? we're catching the exception we're throwing and then re-raising it without doing anything else

Reply:
this is mainly for any exception that might be raised from "result()" since we have seen that if we don't have it in a try/except block then we might not see that exception

Reply:
hmm I think the try should also include ``items.extend(resp.result()["Responses"][self.name])`` since we start accessing result() here